packages/features/calendars/DestinationCalendarSelector.tsx

```diff
@@ -44,31 +50,63 @@ const DestinationCalendarSelector = ({
     return {};
   };
 
+  const SingleValue = (props) => {
+    const { label, subtitle } = props.data;
+    return (
+      <components.SingleValue {...props} className="flex space-x-1">
+        <p>{label}</p> <p className=" text-neutral-500">{subtitle}</p>
+      </components.SingleValue>
+    );
+  };
+
+  const Option = (props) => {
+    const { label } = props.data;
+    return (
+      <components.Option {...props}>
+        <span>{label}</span>
+      </components.Option>
+    );
+  };
+
   useEffect(() => {
+    const { destinationCalendar } = query.data;
+
     const selected = query.data?.connectedCalendars
       .map((connected) => connected.calendars ?? [])
       .flat()
       .find((cal) => cal.externalId === value);
 
     if (selected) {
+      const selectedIntegration = query.data?.connectedCalendars.find((integration) =>
+        integration.calendars?.some((calendar) => calendar.externalId === selected.externalId)
+      );
+
       setSelectedOption({
         value: `${selected.integration}:${selected.externalId}`,
-        label: selected.name || "",
+        label: `${selected.name} ` || "",
+        subtitle: `(${selectedIntegration?.integration.title?.replace(/calendar/i, "")} - ${
+          selectedIntegration?.primary?.name
+        })`,
       });
     }
-  }, [query.data?.connectedCalendars, value]);
+  }, [query.data?.connectedCalendars]);
```

Comment:
Let's display the actual destination calendar rather than "Selected destination calendar"